src/main/js/controls/Graph/Graph.js

```diff
@@ -274,10 +276,20 @@ class Graph extends Construct {
      *  @returns {Graph} - Graph instance
      */
     resize() {
+        let graphHasData = false;
         setCanvasWidth(this.graphContainer, this.config);
         scaleGraph(this.scale, this.config);
         translateGraph(this);
-        this.content.forEach((control) => control.resize(this));
+        removeNoDataView(this.svg);
+        this.content.forEach((control) => {
+            graphHasData =
+                graphHasData ||
+                (control.config.values && !!control.config.values.length);
+            control.resize(this);
+        });
+        if (!graphHasData && this.config.showNoDataText) {
+            drawNoDataView(this.config, this.svg);
```

Comment:
Do you need to create the rect during each resize? 
I would propose just changing the width and positioning during resize and show/hide when the content is available or otherwise. 
Throughout carbon we have a create phase and a transition phase.

Reply:
I have removed the code from resize method and added the create and transition phase.